src/esn.calendar.libs/app/components/event/form/event-form.controller.spec.js

```diff
@@ -2090,5 +2098,89 @@ describe('The CalEventFormController controller', function() {
         //}), newDate.start, newDate.end, scope.event);
       });
     });
+
+    describe('the duplicateEvent function', function() {
+      const eventSkeleton = {
+        start: moment('2018-05-01 10:30'),
+        end: moment('2018-05-01 14:30'),
+        attendees: [{
+          displayName: 'attendee1',
+          email: 'user1@test.com',
+          partstart: 'ACCEPTED'
+        }, {
+          displayName: 'attendee2',
+          email: 'user1@test.com',
+          partstart: 'ACCEPTED'
+        }],
+        sequence: 2, // A property to ignore when copying
+        xOpenpaasVideoconference: undefined // a non defined property
+      };
+
+      beforeEach(function() {
+        scope.event = CalendarShell.fromIncompleteShell(eventSkeleton);
+
+        initController();
+        scope.$digest();
+      });
+
+      it('should create a new copy of the event details correctly', function() {
+        const shellSpy = sinon.spy(CalendarShell, 'fromIncompleteShell');
+
+        scope.duplicateEvent();
+
+        const copiedEvent = shellSpy.firstCall.args[0];
+
+        expect(CalendarShell.fromIncompleteShell).to.have.been.called;
+        // Should ignore properties that cannot be edited in the form.
+        expect(copiedEvent.sequence).to.be.undefined;
+        // Should ignore properties with undefined values
+        expect(copiedEvent.xOpenpaasVideoconference).to.be.undefined;
+      });
+
+      it('should generate a new video conference link if the original event had one', function() {
+        scope.editedEvent = CalendarShell.fromIncompleteShell({
+          ...eventSkeleton,
+          xOpenpaasVideoconference: 'SOMETHING' // an event with a video conference link
+        });
+
+        scope.duplicateEvent();
+
+        expect(VideoConfConfigurationServiceMock.getOpenPaasVideoconferenceAppUrl).to.have.been.called;
+      });
+
+      it('should not generate a new video conference link if the original event didn\'t have one', function() {
+        scope.editedEvent = CalendarShell.fromIncompleteShell(eventSkeleton);
+
+        scope.duplicateEvent();
+
+        expect(VideoConfConfigurationServiceMock.getOpenPaasVideoconferenceAppUrl).to.not.have.been.called;
+      });
+
+      it('should close the previous event form', function() {
+        scope.cancel = sinon.spy();
+        scope.duplicateEvent();
+
+        expect(scope.cancel).to.have.been.called;
+      });
+
+      it('should open a new event form', function() {
+        scope.duplicateEvent();
+        $timeout.flush();
+
+        expect(calOpenEventFormMock).to.have.been.called;
+      });
```

Comment:
I think these two test cases can be merged into one, just "should close the previous event form and open a new event form after a delay".

Reply:
done